edb/schema/casts.py

```diff
@@ -143,7 +144,8 @@ def get_cast_fullname(
         name=sn.get_specialized_name(shortname, *quals))
 
 
-class Cast(s_anno.AnnotationSubject, s_func.VolatilitySubject, s_abc.Cast):
+class Cast(s_anno.AnnotationSubject, s_func.VolatilitySubject, s_abc.Cast,
+           qlkind=qltypes.SchemaObjectClass.CAST):
```

Comment:
Why do you pass `qlkind` to metaclass constructor and not just keep it as a regular class attribute?

Reply:
I think this approach is slightly cleaner since it doesn't imply there's a mutable attribute on a class and we don't need to parse the class dict to validate it.